lib/graphql/sugar/define/resolver.rb

```diff
@@ -3,9 +3,14 @@ module Sugar
     module Define
       module Resolver
         def self.call(type_defn, field_name, type_or_field = nil, desc = nil, **kwargs, &block) # rubocop:disable Metrics/ParameterLists
-          type_or_field ||= kwargs[:type] if !kwargs[:type].nil?
+          # Automatically determine function
+          function_class = Sugar.get_resolver_function(field_name)
+          kwargs[:function] ||= function_class.new
+          kwargs[:resolve] ||= ->(obj, args, ctx) { function_class.new.call(obj, args, ctx) }
+
+          raise "You can define `type` either within #{function_class} or as an argument for #{field_name} field" if kwargs[:function].type && type_or_field
```

Comment:
`You have defined `type` both within #{function_class} and as an argument for #{field_name} field`